assets/js/googlesitekit/api/index.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { deleteItem, getItem, getKeys, setItem } from './cache';
+import { createCacheKey } from './index.private';
+
+// Caching is enabled by default.
+let cachingEnabled = true;
+
+/**
+ * Make a request to a WP REST API Site Kit endpoint.
+ *
+ * @param {string}  type                The data to access. One of 'core' or 'modules'.
+ * @param {string}  identifier          The data identifier, eg. a module slug like `'search-console'`.
+ * @param {string}  datapoint           The endpoint to request data from.
+ * @param {Object}  options             Options to pass to the request
+ * @param {number}  options.cacheTTL    The oldest cache data to use, in seconds.
+ * @param {Object}  options.bodyParams  Request body data to send. (Eg. used for `POST`/`PUT` request variables.)
+ * @param {number}  options.method      HTTP method to use for this request.
+ * @param {Object}  options.queryParams Query params to send with the request.
+ * @param {boolean} options.useCache    Enable or disable caching for this request only. (Caching is only used for `GET` requests.)
+ */
+const siteKitRequest = async ( type, identifier, datapoint, {
```

Comment:
Wouldn't this be better suited in `index.private.js`?

Reply:
So I thought about that, but then thought it might be confusing with the imports—`index.private` would import from `index` and vice-versa. Seems fine though, so I've made the change 👍